File: figuritasFront/src/models/Usuario.js
export default class Usuario {
  constructor({ id, apellido, nombre, fechaNacimiento, mail, direccion, latitud, longitud, maps_link }) {
    this.id = id;
    this.apellido = apellido;
    this.nombre = nombre;
    this.fechaNacimiento = fechaNacimiento;
    this.mail = mail;
    this.direccion = direccion;
    this.latitud = latitud;
    this.longitud = longitud;
    this.maps_link = maps_link;
  }

  get nombreCompleto() {
    return `${this.apellido} ${this.nombre}`;
  }

  get edad() {
    if (!this.fechaNacimiento) return null;
    const nacido = new Date(this.fechaNacimiento);
    if (isNaN(nacido.getTime())) return null;
    const hoy = new Date();
    let edad = hoy.getFullYear() - nacido.getFullYear();
    const m = hoy.getMonth() - nacido.getMonth();
    if (m < 0 || (m === 0 && hoy.getDate() < nacido.getDate())) edad--;
    return edad;
  }
}

File: figuritasBackend/entities/usuario.py
class Usuario:
    def __init__(self, id, apellido, nombre, fecha_nacimiento, mail, direccion, latitud=None, longitud=None, maps_link=None, password=None):
        self.id = id
        self.apellido = apellido
        self.nombre = nombre
        self.fecha_nacimiento = fecha_nacimiento
        self.mail = mail
        self.direccion = direccion
        self.latitud = latitud
        self.longitud = longitud
        self.maps_link = maps_link
        self.password = password
